### PR TITLE
[ci] Re-add running WSU twice in the test suite

### PR DESCRIPTION
**Pull request by sebsoto**
This commit re-adds the WSU running twice in the WSU test suite. The
functionality was mistakingly dropped in commit
65ad3d9d6a6f43e9b9c87b0aba4ab1b676ec814e.

### CODE REVIEW

**Comment by ravisantoshgudimetla** [comment]
/hold

Just to make sure that time we added is enough for tests to pass continuously, let's re-run it manually 2 or 3 times.

**Comment by aravindhp** [comment]
> /hold
> 
> Just to make sure that time we added is enough for tests to pass continuously, let's re-run it manually 2 or 3 times.

5 times :-)

**Comment by sebsoto** [comment]
wsu tests passed 1/1 times

**Comment by sebsoto** [comment]
WSU test passed 1/1 times - rerunning
/test e2e-wsu

File: internal/test/wsu/wsu_test.go [hunk 1]
```diff
@@ -89,35 +89,51 @@ func TestWSU(t *testing.T) {
...
-			setupAndTest(t, i)
+			runPerVMTests(t, framework.WinVMs[i])
 		})
 	}
 
 }
 
-// setupAndTest runs the WSU and runs tests against the setup node
-func setupAndTest(t *testing.T, vmNum int) {
+// runPerVMTests runs all the tests required for a specific VM
```
Comment (aravindhp) [inline internal/test/wsu/wsu_test.go]: Please rename to `runTests()`

File: internal/test/wsu/wsu_test.go [hunk 2]
```diff
@@ -89,35 +89,51 @@ func TestWSU(t *testing.T) {
...
+	// Run our VM test suite
+	runE2ETestSuite(t, vm, wsuStringOutput)
+}
+
+// captureWSULogs saves the WSU logs to the artifact directory
+func captureWSULogs(wsuOut []byte, vm e2ef.WindowsVM, logFileName string) {
 	externalIP := vm.GetCredentials().GetIPAddress()
 	nodeName, err := framework.GetNodeName(externalIP)
 	if err != nil {
 		log.Printf("could not node name associated with the vm %s\n", vm.GetCredentials().GetInstanceId())
```
Comment (aravindhp) [inline internal/test/wsu/wsu_test.go]: Please mention that the logs won't be written to the artifact directory.

File: internal/test/wsu/wsu_test.go [hunk 3]
```diff
@@ -89,35 +89,51 @@ func TestWSU(t *testing.T) {
...
+
+	setupAndTest(t, vm)
+	//Run the test suite again, to ensure that the WSU can be run multiple times against the same VM
+	t.Run("Run the WSU against the same VM again", func(t *testing.T) {
+		setupAndTest(t, vm)
+	})
+}
+
+// setupAndTest runs the WSU and runs tests against the setup node
+func setupAndTest(t *testing.T, vm e2ef.WindowsVM) {
```
Comment (aravindhp) [inline internal/test/wsu/wsu_test.go]: Please reanem to `runWSUandTest()`

File: internal/test/wsu/wsu_test.go [hunk 4]
```diff
@@ -127,7 +127,8 @@ func captureWSULogs(wsuOut []byte, vm e2ef.WindowsVM, logFileName string) {
 	externalIP := vm.GetCredentials().GetIPAddress()
 	nodeName, err := framework.GetNodeName(externalIP)
 	if err != nil {
-		log.Printf("could not node name associated with the vm %s\n", vm.GetCredentials().GetInstanceId())
+		log.Printf("could not node name associated with the vm %s. "+
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply internal/test/wsu/wsu_test.go]: Should have caught that :man_facepalming:

**Comment by sebsoto** [comment]
wsu test passed 1/1 times - rerunning

**Comment by sebsoto** [comment]
test passed 2/2 times - retesting
/test e2e-wsu

**Comment by sebsoto** [comment]
test passed 3/3 times - retesting
/test e2e-wsu

**Comment by sebsoto** [comment]
test passed 4/4 times - retesting
/test e2e-wsu

**Comment by sebsoto** [comment]
Removing hold as I have fufilled the retest request:
/hold cancel

**Pull request merged**